File: unix/boot/spp/rpp/rppfor/outstr.f
      subroutine outstr (str)
      integer str (100)
      integer c
      integer i, j
      i = 1
23000 if (.not.(str (i) .ne. -2))goto 23002
      c = str (i)
      if (.not.(c .ne. 39 .and. c .ne. 34))goto 23003
      call outch (c)
      goto 23004
23003 continue
      i = i + 1
      j = i
23005 if (.not.(str (j) .ne. c))goto 23007
23006 j = j + 1
      goto 23005
23007 continue
      call outnum (j - i)
      call outch (72)
23008 if (.not.(i .lt. j))goto 23010
      call outch (str (i))
23009 i = i + 1
      goto 23008
23010 continue
23004 continue
23001 i = i + 1
      goto 23000
23002 continue
      return
      end

File: unix/boot/spp/rpp/rpprat/outstr.r
#-h-  outstr			  687  local   12/01/80  15:54:32
# outstr - output string; handles quoted literals
   include  defs

   subroutine outstr (str)
   character str (ARB)

   character c

   integer i, j

   for (i = 1; str (i) != EOS; i = i + 1) {
      c = str (i)
      if (c != SQUOTE & c != DQUOTE) {
			 # produce upper case fortran, if desired
	 call outch (c)
	 }
      else {
	 i = i + 1
	 for (j = i; str (j) != c; j = j + 1)	# find end
	    ;
	 call outnum (j - i)
	 call outch (BIGH)
	 for ( ; i < j; i = i + 1)
	    call outch (str (i))
	 }
      }
   return
   end
